pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java

```diff
@@ -686,11 +703,7 @@ public static boolean isSubtypeable(JavaTypeDefinition parameter, JavaTypeDefini
     }
 
     public static JavaTypeDefinition boxPrimitive(JavaTypeDefinition def) {
-        if (!def.isPrimitive()) {
-            return null;
-        }
-
-        return JavaTypeDefinition.forClass(BOXED_PRIMITIVE_SUBTYPE_ORDER.get(PRIMITIVE_SUBTYPE_ORDER.indexOf(def.getType())));
+        return JavaTypeDefinition.forClass(PRIMITIVE_BOXING_RULES.get(def.getType()));
```

Comment:
Ok, I verified: JavaTypeDefinition.forClass deals with null - otherwise, we would still need the above check for primitive.